src/service.py

```diff
@@ -49,24 +51,71 @@ def configure(self, snap_config: dict[str, Any]) -> None:
         self.snap_client.set(snap_config, typed=True)
 
 
-def snap_install(resource: str) -> SnapService:
-    """Install or refresh the snap, and return the snap service.
+def snap_install(resource: Optional[str], channel: str) -> SnapService:
```

Comment:
**nitpick (non-blocking):** 

I have question about this design. The factory function to get the service and install should be separate.

One with side-effect to install the snap. Another is getting the snap service encapsulation, which don't have side-effect.

(Feel free to refactor this part in another PR)

Reply:
We already have the `get_installed_snap_service`, so I think we can remove to return the `SnapService` and this function is only responsible for installing or refresh. Moreover we are never using the response